tools/lint.py

```diff
@@ -0,0 +1,169 @@
+    return negative_filtering(patterns, it)
+
+
+def gather_files(path: str, patterns: list, antipatterns: list):
+    """
+    Gather files, based on `path`, that match `patterns` unix-like specification
+    and do not match `antipatterns` regexes
+    """
+    curr_path = os.getcwd()
+    os.chdir(path)
+    positive_iterators = [glob.iglob(os.path.join('**', pattern), recursive=True) for pattern in
+                          patterns]
+    linted_files = itertools.chain(*positive_iterators)
+    linted_files = (os.path.join(path, file) for file in linted_files)
+    linted_files = negative_filtering(antipatterns.copy(), linted_files)
+    ret = list(linted_files)
+    os.chdir(curr_path)
+    return ret
+
+
+def gen_cmd(dali_root_dir, file_list, process_includes=False):
+    """
+    Command for calling cpplint.py
+    """
+    cmd = ["python",
+           os.path.join(dali_root_dir, "third_party", "cpplint.py"),
+           "--quiet",
+           "--linelength=100",
+           "--root=" + os.path.join(dali_root_dir, "include" if process_includes else "")]
+    cmd.extend(file_list)
+    return cmd
+
+
+def lint(dali_root_dir, file_list, process_includes, n_subproc):
+    """
+    n_subprocesses: how many subprocesses to use for linter processing
+    Returns: 0 if lint passed, 1 otherwise
+    """
+    assert n_subproc > 0
+    if len(file_list)==0:
+        return 0
+    cmds = []
+    diff = int(len(file_list) / n_subproc)
+    for process_idx in range(n_subproc - 1):
+        cmds.append(gen_cmd(dali_root_dir=dali_root_dir,
+                            file_list=file_list[process_idx * diff: (process_idx + 1) * diff],
+                            process_includes=process_includes))
+    cmds.append(gen_cmd(dali_root_dir=dali_root_dir,
+                        file_list=file_list[(n_subproc - 1) * diff:],
+                        process_includes=process_includes))
+    subprocesses = [subprocess.Popen(cmd, stdout=subprocess.PIPE) for cmd in cmds]
+    success = True
+    for subproc in subprocesses:
+        stdout, stderr = subproc.communicate()
+        success *= not bool(subproc.poll())
+        # print(stdout, stderr)
+    return 0 if success else 1
+
+
+def main(dali_root_dir, n_subproc=1, file_list=None):
+    cc_files = gather_files(
+        os.path.join(dali_root_dir, "dali"),
+        ["*.cc", "*.h", "*.cu", "*.cuh"] if file_list is None else file_list,
+        negative_filters)
+    inc_files = gather_files(
+        os.path.join(dali_root_dir, "include"),
+        ["*.h", "*.cuh", "*.inc", "*.inl"] if file_list is None else file_list,
+        negative_filters)
+
+    cc_code = lint(dali_root_dir=dali_root_dir, file_list=cc_files, process_includes=False,
+                   n_subproc=n_subproc)
+    inc_code = lint(dali_root_dir=dali_root_dir, file_list=inc_files, process_includes=True,
+                    n_subproc=n_subproc)
+
+    if cc_code != 0 or inc_code != 0:
+        sys.exit(1)
```

Comment:
```suggestion
         sys.exit(cc_code  + inc_code)
```

Reply:
And no ifs.

Reply:
That would generally by against the doc: https://docs.python.org/3.8/library/sys.html#sys.exit

Reply:
OK